testdata/e2e/compilation_test.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: compilation.example.com
+spec:
+  group: example.com
+  versions:
+    - name: v1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                prop:
+                  type: integer
+                  x-kubernetes-validations:
+                  - rule: 'self == "a string"'
```

Comment:
github is complaining about this end of file. Maybe there is non-visible whitespace?

Reply:
Lack of a trailing newline, maybe? I've made sure all the test files have one now, so I'll mark this as fixed for the time being.